service/src/main/java/org/jboss/aerogear/unifiedpush/service/impl/spring/KeycloakServiceImpl.java

```diff
@@ -397,8 +393,68 @@ public String strip(String fqdn) {
 
 		return StringUtils.EMPTY;
 	}
-	
+
 	public String seperator() {
 		return conf.getRooturlMatcher().seperator();
 	}
+
+    public static final int BULK_SIZE = 100;
+
+    public static final ObjectMapper MAPPER = new ObjectMapper();
+
+    public static final ObjectWriter WRITER = MAPPER.writer();
+
+    @Override
+    public int updateUserAttribute(Map<String, ? extends Collection<UserIdentifiers>> aliasToIdentifiers) {
+        if (!isInitialized()) {
+            throw new RuntimeException("keycloak not initialized");
+        }
+
+        int updated = 0;
+        UsersResource users = this.realm.users();
+        Integer count = users.count();
+        int iterations = count / BULK_SIZE;
+        for (int i = 0; i < iterations; i++) {
+            updated = getUpdated(aliasToIdentifiers, updated, users, i, BULK_SIZE);
+        }
+        int remainder = count % BULK_SIZE;
+        updated = getUpdated(aliasToIdentifiers, updated, users, iterations, remainder);
+
+        return updated;
+    }
+
+    private int getUpdated(Map<String, ? extends Collection<UserIdentifiers>> aliasToIdentifiers, int updated,
+                           UsersResource users, int iterations, int remainder) {
+        List<UserRepresentation> remainderUsers = users.list(iterations * BULK_SIZE, remainder);
+        updated += updateUsers(aliasToIdentifiers, users, remainderUsers);
+        return updated;
+    }
+
+    private int updateUsers(Map<String, ? extends Collection<UserIdentifiers>> aliasToIdentifiers, UsersResource users,
+                            List<UserRepresentation> bulkUsers) {
+        int total = 0;
+        for (UserRepresentation userRepresentation : bulkUsers) {
+            String username = userRepresentation.getUsername();
+            Collection<UserIdentifiers> userIdentifiers = aliasToIdentifiers.get(username);
+            if (userIdentifiers == null) {
+                logger.warn("updateUsers() Found KC user={} without record in cassandra", username);
+                continue;
+            }
+            UserResource userResource = users.get(userRepresentation.getId());
+            try {
```

Comment:
What if _userResource_ is _null_ ? Perhaps logging is in order as well

Reply:
not expected to be null, would be easily identified if will